packages/core/src/common/types.ts

```diff
@@ -15,6 +15,22 @@
 // *****************************************************************************
 
 export type Mutable<T> = { -readonly [P in keyof T]: T[P] };
+export type Nullable<T> = { [P in keyof T]: T[P] | null };
+export type Undefinable<T> = { [P in keyof T]: T[P] | undefined };
+
+/**
+ * Mutates and returns the object so that all ownkeys that are null are made `undefined`
+ */
+export function nullToUndefined<T>(nullable: Nullable<T>): Undefinable<T> {
```

Comment:
I've changed this to make a shallow copy. Since it's only modifying fields that are `null`, that should result in no modifications to objects passed in.